builder/openstack/run_config.go

```diff
@@ -33,6 +33,11 @@ type RunConfig struct {
 	// The name of the base image to use. This is an alternative way of
 	// providing source_image and only either of them can be specified.
 	SourceImageName string `mapstructure:"source_image_name" required:"true"`
+	// The URL of an external base image to use. This is an alternative way of
+	// providing source_image and only either of them can be specified.
+	ExternalSourceImageURL string `mapstructure:"external_source_image_url" required:"true"`
```

Comment:
Is not actually required if SourceImage is set. Setting as required can be confusing because the docs will put it in the required section.

```suggestion
	ExternalSourceImageURL string `mapstructure:"external_source_image_url" required:"false"`
```

Reply:
Yeah, I guess that's a problem with the documentation generator which doesn't seem to support a "requires one of". If you check for `SourceImage`, `SourceImageName` and `SourceImageFilters` they are all marked as required although you must only specify one of them.

Reply:
You're right, that's true! Since the other's are all "required" better leave it like the other's then, like you did.